packages/elevenlabs/src/elevenlabs-transcription-model.ts

```diff
@@ -0,0 +1,270 @@
+            result.include_timestamps = format.include_timestamps ?? undefined;
+          if ('max_segment_chars' in format)
+            result.max_segment_chars = format.max_segment_chars ?? undefined;
+          if ('max_segment_duration_s' in format)
+            result.max_segment_duration_s =
+              format.max_segment_duration_s ?? undefined;
+          if ('segment_on_silence_longer_than_s' in format)
+            result.segment_on_silence_longer_than_s =
+              format.segment_on_silence_longer_than_s ?? undefined;
+          if ('max_characters_per_line' in format)
+            result.max_characters_per_line =
+              format.max_characters_per_line ?? undefined;
+
+          return result;
+        }),
+        file_format: elevenlabsOptions.file_format ?? undefined,
+      };
+
+      for (const key in transcriptionModelOptions) {
+        const value =
+          transcriptionModelOptions[
+            key as keyof ElevenLabsTranscriptionModelOptions
+          ];
+        if (value !== undefined) {
+          formData.append(key, String(value));
+        }
+      }
+    }
+
+    return {
+      formData,
+      warnings,
+    };
+  }
+
+  async doGenerate(
+    options: ElevenLabsTranscriptionCallOptions,
+  ): Promise<Awaited<ReturnType<TranscriptionModelV1['doGenerate']>>> {
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+    const { formData, warnings } = this.getArgs(options);
+
+    const {
+      value: response,
+      responseHeaders,
+      rawValue: rawResponse,
+    } = await postFormDataToApi({
+      url: this.config.url({
+        path: '/v1/speech-to-text',
+        modelId: this.modelId,
+      }),
+      headers: combineHeaders(this.config.headers(), options.headers),
+      formData,
+      failedResponseHandler: elevenlabsFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(
+        elevenlabsTranscriptionResponseSchema,
+      ),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    return {
+      text: response.text,
+      segments:
+        response.words?.map(word => ({
+          text: word.text,
+          startSecond: word.start ?? 0,
+          endSecond: word.end ?? 0,
+        })) ?? [],
+      language: response.language_code,
+      durationInSeconds: response.words?.at(-1)?.end ?? undefined,
+      warnings,
+      response: {
+        timestamp: currentDate,
+        modelId: this.modelId,
+        headers: responseHeaders,
+        body: rawResponse,
+      },
+    };
+  }
+}
+
+const elevenlabsTranscriptionResponseSchema = z.object({
+  language_code: z.string(),
+  language_probability: z.number(),
+  text: z.string(),
+  words: z
+    .array(
+      z.object({
+        text: z.string(),
+        type: z.enum(['word', 'spacing', 'audio_event']),
+        start: z.number().optional(),
+        end: z.number().optional(),
+        speaker_id: z.string().optional(),
```

Comment:
prefer `nullish` over optional (for increased robustness, also below)

Reply:
This is the typed response of Elevenlabs. I can change it, however it would mean modifying the types they've noted in their docs. Let me know if we're good to do that.